### ISSUE TITLE
Tab as CSV-delimiter not supported?

### ISSUE THREAD

**Issue by livarb** [torbengb/bank2ynab#221]
**Describe the bug**
It seems that there is no support for tab as a delimiter. It may also be that the error is caused by something else.

**What did you EXPECT to happen?**
The source files to be parsed.

**What ACTUALLY happened?**
It crashed with the error message below.

**What did you DO? (steps to reproduce)**
Downloaded the most recent version from Git (fork and then clone)
Edited config and tried to parse.

**What's your software environment?**
- MacOS
- Python 3.7.0

**Can you provide other helpful information?**

Tried to add this in the specific config:
Source CSV Delimiter = \t

![image](https://user-images.githubusercontent.com/3617256/44946986-8e958300-ae06-11e8-9618-f57970f09d38.png)

**Attachments**
If applicable, add sample files or screenshots to help explain your problem.

Please let me know if you want more details.

**Comment by toyg** [comment]
the csv module works fine with tabs. The bug is likely to be the config machinery not picking up the character properly.

**Comment by nocalla** [comment]
I suspect this is down to the ConfigParser - I ran into something similar with % meaning we had to change to RawConfigParser instead of the default.

**Comment by nocalla** [comment]
[This](https://stackoverflow.com/questions/5753332/passing-meta-characters-to-python-as-arguments-from-command-line) seems to be what's happening here - the "\t" is being interpreted literally as "\" & "t" instead of being parsed into a tab. 

We could force an evaluation of the delimiter in fix_conf_params as a solution maybe? Or there seems to be a "callback" option outlined in the link above, but that might be a bit complex for the issue at hand.

**Comment by toyg** [comment]
The callback route is the cleaner one, because it offers a single place where to store all parsing-related hacks you might have now and in future. It's also safer: what if I put the python equivalent of "format c:" in that field, and you eval it? :)

**Comment by nocalla** [comment]
That's a fair point! Let's keep our hard drives close to bursting.

**Comment by nocalla** [comment]
I don't understand how the callback works, but I've pushed a quick and dirty fix to #222 to fix this particular case. 

```
if config["input_delimiter"] == "\\t":
        config["input_delimiter"] = "\t"
```

I've also added a tab test case to our unit tests.

@livarb, can you try out the branch over at #222 to see if it works for you?
